File: fibonacci.js
function fb(i) {
    var element; 
    var memo = []
    if( i == 0 ) {
        return 0; 
    } else if ( (i == 1) || (i == 2) ) {
        return  1;
    } else if (memo[i] != 0) {
        return memo[i];  
    } else {
        memo[i] =  fb(i-2) + fb(i-1) ; 
        return memo[i];
    }
}
 console.log( fb(6) )

File: dynamicprograming/fibonacci.py
#  dynamic programing 
# memoized solution 
def fib(n, memo):
    if memo[n] != null:
        return memo[n]
    if n == 1 or n == 2: 
        result = 1
    else: 
        result = fib(n-1) + fib(n-2)
    memo[n] = result
    return result

# bootom-up approch

def fib_bottom_up(n):
    if n == 1 or n == 2:
        return 1
    bootom_up = new int[n + 1]
    bootom_up[1] = 1
    bootom_up[2] = 1

    for i from 3 upto n: 
        bootom_up[i] = bootom_up[i-1] + bootom_up[i-2]
    return bootom_up[n]
